lib/awspec/type/resource_base.rb

```diff
@@ -1,9 +1,10 @@
 module Awspec::Type
   class ResourceBase < Base
-    attr_reader :id, :resource_via_client
+    attr_reader :id, :resource_via_client, :params
 
-    def initialize(display_name = nil)
-      @display_name = display_name
+    def initialize(params)
+      @params = params
+      @display_name = @params.is_a?(Hash) ? @params[:name] : params
```

Comment:
This look like https://github.com/k1LoW/awspec/pull/225/files#diff-f28a4620f1d9da6f3df937c097fdea8dR5
( link #224  #225 )